inc/ctrs_leak_check.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef CTRS_LEAK_CHECK_H
+#define CTRS_LEAK_CHECK_H
+
+#ifdef __cplusplus
+#include <cstddef>
+#else
+#include <stddef.h>
+#endif
+
+#include "testrunnerswitcher.h"
+
+#ifdef VLD_OPT_REPORT_TO_STDOUT
+// MAIN_RUN_TEST_SUITE_WITH_LEAK_CHECK macro with VLD leak checks
+#define MAIN_RUN_TEST_SUITE_WITH_LEAK_CHECK(test_suite) \
+int main(void) \
+{ \
+    size_t failedTestCount = 0; \
+    VLD_UINT initial_leak_count = VLDGetLeaksCount(); \
+    RUN_TEST_SUITE(test_suite, failedTestCount); \
```

Comment:
>RUN_TEST_SUITE [](start = 4, length = 14)

But why not have the vld.h and its glorious functionality under RUN_TEST_SUITE macro?

Reply:
Initially this was a macro in the consumer repo (in test_helpers). But then test_helpers would need to be added to all CMakeLists. So I decided to move the macro to ctestrunnerswitcher.

But I agree, having it under RUN_TEST_SUITE is much better.

---
In reply to: [291299922](https://github.com/Azure/azure-c-testrunnerswitcher/pull/19#discussion_r291299922) [](ancestors = 291299922)